jetty-osgi/jetty-osgi-boot/pom.xml

```diff
@@ -82,7 +82,7 @@
               jakarta.mail.internet;version="2.0";resolution:=optional,
               jakarta.mail.search;version="2.0";resolution:=optional,
               jakarta.mail.util;version="2.0";resolution:=optional,
-              jakarta.servlet;version="[$(version;==;${}),$(version;+;${jetty.servlet5.api.version}))",
+              jakarta.servlet;version="[$(version;==;${jetty.servlet5.api.version}),$(version;+;${jetty.servlet5.api.version}))",
```

Comment:
Something seems wrong here: this refers to `jetty.servlet5.api.version`, yet line 22 above refers to `org.eclipse.jetty.servlet6`.

Reply:
There is no servlet 6 version available here yet, it is still using the Servlet 5 version even in `jetty-webapp` which we are importing on line 22. The versions should be updated to servlet6 when it becomes available, but if we changed this now to `jetty.servlet6.api.version` it would fail.

Reply:
@lachlan-roberts you can locally build the servlet6 api and build against that. That's what the old jetty-12-experimental branch was doing. The servlet api is gearing up for a release of 6 soon, so it would be good to have these things in place.

Reply:
Sure, but I think it would be best to do the update to servlet 6 version in a different PR to this. This is more aimed at the restructuring of the things under `jetty-servlet-parent` so that the project would build.

I just fixed this line because it was causing a failing test, it was brought over from 11 incorrectly.

Reply:
@lachlan-roberts also, I'm not sure that having split packages is a good idea. Eg both `jetty-webapp` and `jetty-webapp-common` are `org.eclipse.jetty.webapp`.

Reply:
@lachlan-roberts I think you should be consistent. If you were just porting servlet5 to jetty12, then you should have called the modules `servlet5` not `servlet6`.  But, you've named things `servlet6`. So I suggest you also rename these properties to `servlet6` and just leave them pointing to servlet 5 jars.

Reply:
> I'm not sure that having split packages is a good idea. Eg both jetty-webapp and jetty-webapp-common are org.eclipse.jetty.webapp.

The idea is that we have `servlet6/jetty-webapp` which might be able to share some code with `servlet5/jetty-webapp` and the shared code would go in `jetty-webapp-common`. But currently there is nothing in these modules as we don't have multiple implementations yet. All the shared modules are currently empty so I could remove them in this PR, but they'll probably have to be brought back eventually. Maybe the package names in `servlet6/jetty-webapp` will need to be `org.eclipse.jetty.webapp6`.

> I think you should be consistent. If you were just porting servlet5 to jetty12, then you should have called the modules servlet5 not servlet6. But, you've named things servlet6. So I suggest you also rename these properties to servlet6 and just leave them pointing to servlet 5 jars.

I think we're going for a complete servlet 6 implementation before we try to support the multiple versions so thats why I called it servlet6 and not servlet5. I can start renaming properties and things here to servlet6, I just thought they'd be lost within the 500+ file changes and will be hard to review. I'll give it a go to see how much there is to change.

Reply:
@janbartel I have updated all the properties to `jetty.servlet6.api.version`.